### ISSUE TITLE
Enhance Kinesis Data Loader: Enable Block-Specific AWS Credentials/ARNs in YAML for Accessing External Kinesis Streams"

### ISSUE THREAD

**Issue by xylesoft** [mage-ai/mage-ai#4287]
**Is your feature request related to a problem?**
Yes, the current limitation in Mage.ai is that it only uses system-wide environmental variables to store AWS credentials. This design prevents me from effectively ingesting Kinesis data streams into my Mage setup, as my Kinesis resources are spread across multiple AWS accounts and regions, separate from the Mage.ai AWS account.

**Describe the solution you'd like**
I propose extending the KinesisConfig class within Mage.ai to include fields for AWS credentials (aws_access_key_id, aws_secret_access_key, and optionally aws_session_token). This enhancement will allow AWS credentials to be specified directly in the YAML configuration for each Kinesis Data Loader streaming block. Consequently, the KinesisSource class's init_client method should be updated to utilize these credentials when initializing the boto3 Kinesis client. This change will provide the flexibility to configure multiple Kinesis blocks with different AWS credentials, allowing Mage.ai to ingest data from Kinesis streams located in various AWS accounts and regions.

**Describe alternatives you've considered**
One alternative could be to manually manage multiple sets of environmental variables and switch between them when accessing different AWS resources. However, this approach is cumbersome, error-prone, and not scalable for a system that needs to interact with multiple AWS resources across different accounts and regions.

**Additional context**
This feature is crucial for environments where Kinesis streams are not centralized in a single AWS account or region. By allowing Mage.ai to use AWS credentials specified for each Kinesis block directly, we can significantly enhance its adaptability and usability in more complex AWS setups. This change also aligns with best practices in cloud resource management, where resources can be distributed across multiple accounts for security, cost management, and organizational structuring.

**Comment by wangxiaoyou1993** [comment]
@xylesoft are you still interested in contributing?

**Comment by xylesoft** [comment]
> @xylesoft are you still interested in contributing?

Yes, apologies I've been on vacation recently and also busy with work. I've started looking into the issue.
